### ISSUE TITLE
Use yaml library to write cloud-init's user-data

### ISSUE THREAD

**Issue by skobyda** [cockpit-project/cockpit-machines#1235]
At `src/scripts/install_machine.py` we write cloud-init's user-data yaml file ourself. Perhaps this could be simplified by using some library for that.

Proposed by @jelly at https://github.com/cockpit-project/cockpit-machines/pull/1150#discussion_r1343814396

**Comment by martinpitt** [comment]
I'm not a fan. Writing YAML is super easy, we really don't need Yet Another Dependency for that. YAML is (rather) hard to parse, but we don't have to do that.

**Comment by jelly** [comment]
Let's close this then.
